virttest/utils_test/__init__.py

```diff
@@ -1839,10 +1845,85 @@ def app_running(self):
 
+    def _git_download(self, url, destination):
+        """
+        git download
+        :param url: download url
+        :param destination: download destination
+        """
+        from avocado.utils import git
+        destination = os.path.join(destination, self.base_name)
+        git.get_repo(url, destination_dir=destination)
+
+    def _tarball_download(self, url, destination):
+        """
+        tarball download and uncompress
+        :param url: dowload url
+        :param destination: download destination
+        """
+        import tarfile
+        tarball_path = os.path.join(destination, self.base_name)
+        download.get_file(url, tarball_path)
+        tarball = tarfile.open(tarball_path)
+        tarball.extractall(destination)
+        self.base_name = tarball.members[0].name.split("/")[0]
+        tarball.close()
+        self.session.cmd_output_safe("rm -f %s" % tarball_path)
+
+    @error_context.context_aware
+    def download_stress(self):
+        """
+        Download stress tool
+        """
+        url = self.download_url
+        tmp_path = data_dir.get_tmp_dir()
+
+        error_context.context('Download stress tool from %s' % url,
+                              logging.info)
+        download_method = getattr(self, "_%s_download" % self.download_type)
+        download_method(url, tmp_path)
+
+        error_context.context('Copy stress tool to guest', logging.info)
+        self.vm.copy_files_to(os.path.join(tmp_path, self.base_name), self.dst_path)
+
+    @error_context.context_aware
+    def install(self):
+        """
+        To download, abstract, build and install the stress tool
+        """
+        self.download_stress()
+        install_path = os.path.join(self.dst_path, self.base_name)
+        self.session.cmd_output_safe('cd %s' % install_path)
+        error_context.context('make and install the %s' %
+                              self.stress_type, logging.info)
+        self.session.cmd_output_safe(self.make_cmds)
+
+    @error_context.context_aware
+    def clean(self):
+        """
+        Uninstall stress application, and clean the source files
+        """
+        session = self.session
+        install_path = os.path.join(self.dst_path, self.base_name)
+        if session.cmd_status('cd %s' % install_path) != 0:
+            error_context.context("No source files found in path %s" % path)
+            session.close()
+            return
+
+        uninstall_cmd = './configure && make uninstall'
+        error_context.context('Uninstall %s' % self.stress_type, logging.info)
+        status, output = session.cmd_status_output(uninstall_cmd)
+        if status != 0:
+            error_context.context('Uninstall stress failed with '
+                                  'error: %s' % output, logging.error)
+        error_context.context('Remove the source files', logging.info)
+        rm_cmd = 'cd && rm -rf %s' % install_path
+        session.cmd_output_safe(rm_cmd)
```

Comment:
@huangyum I supposed we do not need, because I download it to the temporary path `data_dir.get_tmp_dir()`, I think it will be removed by the environment.

Reply:
we can use `data_dir.clean_tmp_files()`

Reply:
@balamuruhans Maybe other cases will use this temporary directory at the same time, I am afraid that I will delete the files that still under use. Avocado will execute it at the end of its run() function, I suppose that it would be better to clean them after all cases finished.

Reply:
Agreed! Thank you.